src/account/operations/syncing/foundries.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::{HashMap, HashSet};
+
+use iota_client::{
+    api_types::response::OutputWithMetadataResponse,
+    block::output::{FoundryId, Output},
+};
+
+use crate::account::handle::AccountHandle;
+
+impl AccountHandle {
+    pub(crate) async fn request_and_store_foundry_outputs(&self, foundry_ids: HashSet<FoundryId>) -> crate::Result<()> {
+        log::debug!("[SYNC] request_and_store_foundry_outputs");
+        let mut foundries: HashMap<FoundryId, OutputWithMetadataResponse> =
+            self.read().await.native_token_foundries().clone();
+
+        let mut tasks = Vec::new();
+
+        for foundry_id in foundry_ids {
+            // Don't request known foundries again
+            if foundries.contains_key(&foundry_id) {
+                continue;
+            }
+
+            let client = self.client.clone();
+            tasks.push(async move {
+                tokio::spawn(async move {
+                    match client.foundry_output_id(foundry_id).await {
+                        Ok(output_id) => Ok(Some(client.get_output(&output_id).await?)),
+                        Err(iota_client::Error::NotFound(_)) => Ok(None),
+                        Err(e) => Err(crate::Error::Client(e.into())),
+                    }
+                })
+                .await
+            });
+        }
+
+        let results = futures::future::try_join_all(tasks).await?;
+        // Update account with new foundries
+        let token_supply = self.client.get_token_supply().await?;
+        for res in results {
+            if let Some(foundry_output_with_metadata) = res? {
+                let output = Output::try_from_dto(&foundry_output_with_metadata.output, token_supply)?;
+                if let Output::Foundry(foundry) = output {
+                    foundries.insert(foundry.id(), foundry_output_with_metadata);
```

Comment:
Do we ever remove foundries or it's an ever growing list ?

Reply:
Currently it's an ever growing list which could also be useful if you want to know the metadata for a native token for which you had an output that got spent already (I guess if one wants to remove foundries, it would happen if you have no unspent output with this native token id anymore)
Also it shouldn't be in an issue? Of course you can spam an address with native tokens to increase the storage size, but this also happens for outputs without without native tokens, so it's not a new created potential issue